php_memcached_private.h

```diff
@@ -48,11 +48,11 @@
 #include <ext/standard/basic_functions.h>
 
 #ifdef PHP_WIN32
-  #  if PHP_VERSION_ID >= 80000
-  #  include "php_stdint.h"
-#else
-# include "win32/php_stdint.h"
-#endif
+  #if (defined(_MSC_VER) && _MSC_VER<=1920) 
```

Comment:
reference [PHP 8 compatibility Windows](https://github.com/phpredis/phpredis/pull/1851/files)

Reply:
[for this commit](https://github.com/php/php-src/commit/d08451b2ca816e3c8f7c07888355a926bef4a487)

Reply:
Found it: https://github.com/php/php-src/blob/PHP-8.2/UPGRADING.INTERNALS#L26

But I still don't understand why the MSC_VER is the right way to determine whether to use the win32 header variant. Isn't the issue here that the PHP version needs to be checked for >= 80200?

Reply:
Yeah, it's probably best to check for the PHP version only; in theory, `<stdint.h>` might not be available (i.e. for very old MSVC/SDKs), but in practice this can be ignored.

Reply:
[参考这个](https://github.com/phpredis/phpredis/pull/1851)

Reply:
The relevant point here is that `win32/php_stdint.h` can be used only prior to PHP 8.2.0. After that version, a "native" `stdint.h` has to be used, regardless of the compiler (which could be almost any version of MSVC, but also clang, for example). As is, someone using Visual Studio 2019 version 16.0.0 or earlier would not be able to build the extension for PHP 8.2, although that should work.

The clean solution would be to check for the availability of stdint.h in config.w32, but there is no readily usable support for that (unlike with autoconf). So the next best solution is likely to rely on `win32/php_stdint.h` prior to PHP 8.2.0, and to just assume `stdint.h` is available as of PHP 8.2.0.

Reply:
Oh, I just realize that this is actually about `win32/php_stdint.h` which is removed as of PHP 8.0.0 (I thought it is about `main/php_stdint.h` which is removed as of PHP 8.2.0 only). So, I don't understand why there should be any change at all. Apparently, [standard Windows builds for PHP 8.0 and 8.1 are successful](https://github.com/php-memcached-dev/php-memcached/actions/runs/3901598618/jobs/6663642162).

Reply:
I want to less if else condition

Reply:
Okay, simplification is nice, and compatibility with PHP 8.2.0 is important. But why not:
````diff
 php_memcached_private.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/php_memcached_private.h b/php_memcached_private.h
index b4b1115..2d67e36 100644
--- a/php_memcached_private.h
+++ b/php_memcached_private.h
@@ -48,11 +48,11 @@
 #include <ext/standard/basic_functions.h>
 
 #ifdef PHP_WIN32
-  #  if PHP_VERSION_ID >= 80000
-  #  include "php_stdint.h"
-#else
-# include "win32/php_stdint.h"
-#endif
+# if PHP_VERSION_ID >= 80000
+#  include <stdint.h>
+# else
+#  include "win32/php_stdint.h"
+# endif
 #else
 /* Used to store the size of the block */
 #  if defined(HAVE_INTTYPES_H)
````

This is basically the same as your suggestion, but checking for `PHP_VERSION` instead of `_MSC_VER`.

Reply:
ok,good suggestion.i'll modify it.

Reply:
thank you, I'm keeping an eye out for the PR update!